jwst/extract_1d/extract.py

```diff
@@ -2969,14 +2959,19 @@ def do_extract1d(input_model, ref_dict, smoothing_length=None,
                 spec.slit_dec = dec
                 spec.spectral_order = sp_order
                 spec.dispersion_direction = extract_params['dispaxis']
-                if slitname is not None and slitname != "ANY":
-                    spec.name = slitname
+                # The first argument of copy_keyword_info was originally
+                # intended to be a SlitModel object, but ImageModel now has
+                # the attributes we will look for in this function.
+                copy_keyword_info(input_model, slitname, spec)
                 output_model.spec.append(spec)
 
         elif isinstance(input_model, (datamodels.CubeModel,
                                       datamodels.SlitModel)):
 
             slit = None
+            if hasattr(input_model, "name") and input_model.name is not None:
```

Comment:
```suggestion
            if getattr(input_model, "name", default=None) is not None:
```

I've seen this bit a couple of times now, what does "else slit name was assigned above" mean here?

Could also just assign `slitname` directly perhaps: 
```python
slitname = getattr(input_model, 'name')
```

Reply:
@jwhite3 Those are all good suggestions; thanks.